a3fe/tests/test_run_integration.py

```diff
@@ -202,11 +202,39 @@ def test_slurm_adaptive_run(self, slurm_calc_adaptive, system_prep_config):
         # Check that the calculation is not running
         assert not calc.running
 
-        # Check that the equilibration time has successfully been set
-        for leg in calc.legs:
-            for stage in leg.stages:
-                for lam_win in stage.lam_windows:
-                    assert lam_win.equil_time is not None
+        # Check that the equilibration time has successfully been set and matches expected value
+        for leg_name in LEGS_WITH_STAGES.keys():
+            for stage_name in LEGS_WITH_STAGES[leg_name]:
+                # Check that the check_equil_multiwindow_paired_t.txt file exists
+                leg_path = os.path.join(calc.base_dir, leg_name)
+                stage_path = os.path.join(leg_path, stage_name)
```

Comment:
This looks great. Just commenting on something I've done badly - I've mixed the using strings as paths with os.path.join and using pathlib.Path throughout the code. I think pathlib.Path is nicer and if I wrote from scratch I would only use pathlib.Path. Just mentioning for awareness though - this isn't something we really need to fix.